transcom/mymove
# MAIN B-21578 - Add overall weight limit & Add TOO, TIO, Prime Functions

Upstream story: https://github.com/transcom/mymove/pull/14206

## [B-21578](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21578)

INT PR: https://github.com/transcom/mymove/pull/14271

## Summary

This sets overall weight limit to 350 lb when the orders type is `Student Travel`. You should see this limit as a TOO in move task order page.

- Updated the `SetWeightAllotment` func to take in additional param (ordersType).
- Added `studentTravel` as weightAllotment and set weight limit to 350.

### How to test

1. In your `.envrc` file -> change to `export FEATURE_FLAG_ENABLE_ALASKA=true` to turn on Alaska FF locally.
2. Close your server -> run `direnv allow` and then restart your server.
3. As a customer -> create a HHG move with `Student Travel` as your Orders type.
4. Go through the normal flow to approve as TOO and have move task order page available.
5. In Move task order page -> under "WEIGHT ALLOWANCE" you should see 350 lbs.
6. As Prime -> find the move and click "Update Shipment".
7. Put 555 lbs for Estimated and Actual weight -> Pickup and delivery date needs to be in the future.
8. After updating, under Shipments -> "Reweigh Weight: **Missing**" should show -> click "Edit" link next to it.
9. Put 555 for Reweigh Weight -> any remarks -> Save.
10. As TOO -> go to Move task order -> yellow flag about "This move is at risk for excess weight.  Review billable weight" should show up.
11. Click "Review billable weight" and put 350 lbs and save.
12. As TIO -> go to Payment request tab -> you should see "Move exceeds max billable weight" warning in red text next to Billable weights.

Check if new orders types are showing in Prime Api:
1. Prime sim -> find the move you created above.
2. Open dev tools -> Network tab -> refresh page -> find move-task-order GET request (starts with `http://officelocal:3000/prime/v3/move-task-orders/`).
3. Preview data -> under `Orders` you should see `ordersType` with "STUDENT_TRAVEL" as the value.


## Screenshots
![image](https://github.com/user-attachments/assets/a33f718f-8055-4c00-9937-8634fde3a87e)
![image](https://github.com/user-attachments/assets/97485c84-7d3f-483f-a34a-3dfc7d99723a)
![image](https://github.com/user-attachments/assets/ac38ae1b-10be-4567-ba5c-b2c4e6b74066)
